modules/backend/controllers/Auth.php

```diff
@@ -167,32 +168,28 @@ public function restore_onSubmit()
         ];
 
         $validation = Validator::make(post(), $rules);
-        if ($validation->fails()) {
-            throw new ValidationException($validation);
-        }
-
-        $user = BackendAuth::findUserByLogin(post('login'));
-        if (!$user) {
-            throw new ValidationException([
-                'login' => trans('backend::lang.account.restore_error', ['login' => post('login')])
-            ]);
+        if (!$validation->fails()) {
+            /** Search for a backend user with the specified username. */
+            $user = BackendAuth::findUserByLogin(post('login'));
+
+            if ($user) {
+                $code = $user->getResetPasswordCode();
+                $link = Backend::url('backend/auth/reset/' . $user->id . '/' . $code);
+
+                $data = [
+                    'name' => $user->full_name,
+                    'link' => $link,
+                ];
+
+                Mail::send('backend::mail.restore', $data, function ($message) use ($user) {
+                    $message->to($user->email, $user->full_name)->subject(trans('backend::lang.account.password_reset'));
+                });
+            }
         }
 
         Flash::success(trans('backend::lang.account.restore_success'));
 
-        $code = $user->getResetPasswordCode();
-        $link = Backend::url('backend/auth/reset/' . $user->id . '/' . $code);
-
-        $data = [
-            'name' => $user->full_name,
-            'link' => $link,
-        ];
-
-        Mail::send('backend::mail.restore', $data, function ($message) use ($user) {
-            $message->to($user->email, $user->full_name)->subject(trans('backend::lang.account.password_reset'));
-        });
-
-        return Backend::redirect('backend/auth/signin');
+        return Redirect::refresh();
```

Comment:
What's your thought process behind refreshing the current reset page vs redirecting to the login page?

Reply:
To allow the user to quickly reenter the username in case he/she doesn't type the correct username and the password reset email is not received.

Reply:
To allow the user to quickly reenter the username in case he/she doesn't type the correct username and the password reset email is not received.